tests/hlsl/dxsdk/FluidCS11/FluidRender.hlsl

```diff
@@ -71,7 +70,7 @@ float4 VisualizeNumber(float n, float lower, float upper)
 
 VSParticleOut ParticleVS(uint ID : SV_VertexID)
 {
-    VSParticleOut Out = (VSParticleOut)0;
```

Comment:
I think we have an open issue for supporting this (awful) syntax from HLSL. Ideally we should add a simpler way to get default initialization (like just using `{}`) and keep this kind of cast as an error.

If we *need* to support this case, then I'd end up wanting to make it something we special-case so that it is only supported inside of `.hlsl` files, and not `.slang` files, and then warn loudly about it. :)